pkg/framework/plugins/nodeutilization/lownodeutilization.go

```diff
@@ -19,35 +19,65 @@ package nodeutilization
 import (
 	"context"
 	"fmt"
-
 	v1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/resource"
-	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/klog/v2"
-
-	"sigs.k8s.io/descheduler/pkg/api"
+	"sigs.k8s.io/descheduler/pkg/apis/componentconfig"
 	"sigs.k8s.io/descheduler/pkg/descheduler/evictions"
 	nodeutil "sigs.k8s.io/descheduler/pkg/descheduler/node"
 	podutil "sigs.k8s.io/descheduler/pkg/descheduler/pod"
+	"sigs.k8s.io/descheduler/pkg/framework"
 )
 
+const LowNodeUtilizationPluginName = "LowNodeUtilization"
+
 // LowNodeUtilization evicts pods from overutilized nodes to underutilized nodes. Note that CPU/Memory requests are used
 // to calculate nodes' utilization and not the actual resource usage.
-func LowNodeUtilization(ctx context.Context, client clientset.Interface, strategy api.DeschedulerStrategy, nodes []*v1.Node, podEvictor *evictions.PodEvictor, evictorFilter *evictions.EvictorFilter, getPodsAssignedToNode podutil.GetPodsAssignedToNodeFunc) {
-	// TODO: May be create a struct for the strategy as well, so that we don't have to pass along the all the params?
-	if err := validateNodeUtilizationParams(strategy.Params); err != nil {
-		klog.ErrorS(err, "Invalid LowNodeUtilization parameters")
-		return
+
+type LowNodeUtilization struct {
+	handle     framework.Handle
+	args       *componentconfig.LowNodeUtilizationArgs
+	podFilter  func(pod *v1.Pod) bool
+	podEvictor *evictions.PodEvictor
+}
+
+var _ framework.Plugin = &LowNodeUtilization{}
+var _ framework.BalancePlugin = &LowNodeUtilization{}
+
+// NewLowNodeUtilization builds plugin from its arguments while passing a handle
+func NewLowNodeUtilization(args runtime.Object, handle framework.Handle, podEvictor *evictions.PodEvictor) (framework.Plugin, error) {
```

Comment:
All plugin constructors are expected to accept only `args` and `handle` parameters. Pod evictor is to be retrieved from the `handle` as `d.handle.Evictor()`. Is there a reason to the third `podEvictor` parameter?

Reply:
i will click resolve here, and discuss on below